usaspending_api/disaster/v2/views/award/amount.py

```diff
@@ -0,0 +1,98 @@
+from django.db.models import Q, Sum, Count, F, Value, Case, When
+from django.db.models.functions import Coalesce
+from rest_framework.response import Response
+
+from usaspending_api.awards.models import FinancialAccountsByAwards
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.disaster.v2.views.disaster_base import DisasterBase, AwardTypeMixin
+
+
+def submission_window_cutoff(min_date, monthly_sub, quarterly_sub):
+    sub_queries = []
+    if monthly_sub:
+        sub_queries.append(
+            Q(
+                Q(submission__quarter_format_flag=False)
+                & Q(submission__reporting_period_end__lte=monthly_sub["submission_reveal_date"])
+            )
+        )
+
+    sub_queries.append(
+        Q(
+            Q(submission__quarter_format_flag=True)
+            & Q(submission__reporting_period_end__lte=quarterly_sub["submission_reveal_date"])
+        )
+    )
+
+    sub_queryset = sub_queries.pop()
+    for query in sub_queries:
+        sub_queryset |= query
+
+    return [
+        Q(submission__reporting_period_start__gte=min_date),
+        Q(sub_queryset),
+    ]
+
+
+class AmountViewSet(AwardTypeMixin, DisasterBase):
+    """View to implement the API"""
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/disaster/award/amount.md"
+
+    @cache_response()
+    def post(self, request):
+        return Response(self.queryset)
+
+    @property
+    def queryset(self):
+        filters = [
+            Q(award__type__in=self.award_type_codes),
+            Q(award__isnull=False),
+            Q(disaster_emergency_fund__in=self.def_codes),
+            Q(treasury_account__federal_account__isnull=False),
+            Q(treasury_account__isnull=False),
+        ]
+        filters.extend(
+            submission_window_cutoff(
+                self.reporting_period_min,
+                self.last_closed_monthly_submission_dates,
+                self.last_closed_quarterly_submission_dates,
+            )
+        )
+        case_when_queries = []
+        if self.last_closed_monthly_submission_dates:
+            case_when_queries.append(
+                Q(
+                    submission__reporting_fiscal_year=self.last_closed_monthly_submission_dates[
+                        "submission_fiscal_year"
+                    ],
+                    submission__reporting_fiscal_period=self.last_closed_monthly_submission_dates[
+                        "submission_fiscal_month"
+                    ],
+                    submission__quarter_format_flag=False,
+                )
+            )
+
+        case_when_queries.append(
+            Q(
+                submission__reporting_fiscal_year=self.last_closed_quarterly_submission_dates["submission_fiscal_year"],
+                submission__reporting_fiscal_quarter=self.last_closed_quarterly_submission_dates[
```

Comment:
Is this (and monthly above) correct?  Should we be summing `transaction_obligated_amount` for the year up to and including last closed period?  And should `gross_outlay_amount_by_award_cpe` be following that crazy "sum all previous final of fy values" logic?

Reply:
Not sure. This is copy-paste from other code which was (hopefully) vetted and should basically be the ORM you and I worked on last week

Reply:
Fixed code in another PR